Arm backend: Add 16A8W support and test for sigmoid operation

Ninja91 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #13802
* #13801
* #13800
* #13799
* #13798
* #13797
* __->__ #13796

Add 16A8W quantization support and test for the sigmoid operation in ExecutorTorch ARM backend.

This follows the pattern established for linear and mul operations, extending int16 support to sigmoid operations.

Changes:
- Add INT16 dtype validation support in op_sigmoid.py
- Add test_sigmoid_tensor_16a8w_tosa_INT test function
- Enable test_sigmoid.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80510729](https://our.internmc.facebook.com/intern/diff/D80510729/)

cc @digantdesai @freddan80 @per @zingo @oscarandersson8218

Ninja91 committed:
Add 16A8W support and test for sigmoid operation

Add 16A8W quantization support and test for the sigmoid operation in ExecutorTorch ARM backend.

This follows the pattern established for linear and mul operations, extending int16 support to sigmoid operations.

Changes:
- Add INT16 dtype validation support in op_sigmoid.py
- Add test_sigmoid_tensor_16a8w_tosa_INT test function
- Enable test_sigmoid.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80510729](https://our.internmc.facebook.com/intern/diff/D80510729/)

[ghstack-poisoned]

digantdesai commented on backends/arm/test/ops/test_sigmoid.py:
```diff
@@ -253,3 +260,102 @@ def test_sigmoid_vgf_INT_add_3():
+
+
+@common.parametrize("test_data", test_data_suite)
+def test_sigmoid_16a8w_tosa_INT(test_data: torch.Tensor):
```
no xfail ==> passing?

Ninja91 committed:
Update on "Arm backend: Add 16A8W support and test for sigmoid operation"


Add 16A8W quantization support and test for the sigmoid operation in ExecutorTorch ARM backend.

This follows the pattern established for linear and mul operations, extending int16 support to sigmoid operations.

Changes:
- Add INT16 dtype validation support in op_sigmoid.py
- Add test_sigmoid_tensor_16a8w_tosa_INT test function
- Enable test_sigmoid.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80510729](https://our.internmc.facebook.com/intern/diff/D80510729/)

cc digantdesai freddan80 per zingo oscarandersson8218

[ghstack-poisoned]

Ninja91 committed:
Update on "Arm backend: Add 16A8W support and test for sigmoid operation"


Add 16A8W quantization support and test for the sigmoid operation in ExecutorTorch ARM backend.

This follows the pattern established for linear and mul operations, extending int16 support to sigmoid operations.

Changes:
- Add INT16 dtype validation support in op_sigmoid.py
- Add test_sigmoid_tensor_16a8w_tosa_INT test function
- Enable test_sigmoid.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80510729](https://our.internmc.facebook.com/intern/diff/D80510729/)

cc digantdesai freddan80 per zingo oscarandersson8218

[ghstack-poisoned]

Ninja91 committed:
Update on "Arm backend: Add 16A8W support and test for sigmoid operation"


Add 16A8W quantization support and test for the sigmoid operation in ExecutorTorch ARM backend.

This follows the pattern established for linear and mul operations, extending int16 support to sigmoid operations.

Changes:
- Add INT16 dtype validation support in op_sigmoid.py
- Add test_sigmoid_tensor_16a8w_tosa_INT test function
- Enable test_sigmoid.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80510729](https://our.internmc.facebook.com/intern/diff/D80510729/)

cc digantdesai freddan80 per zingo oscarandersson8218

[ghstack-poisoned]

Ninja91 committed:
Update on "Arm backend: Add 16A8W support and test for sigmoid operation"


Add 16A8W quantization support and test for the sigmoid operation in ExecutorTorch ARM backend.

This follows the pattern established for linear and mul operations, extending int16 support to sigmoid operations.

Changes:
- Add INT16 dtype validation support in op_sigmoid.py
- Add test_sigmoid_tensor_16a8w_tosa_INT test function
- Enable test_sigmoid.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80510729](https://our.internmc.facebook.com/intern/diff/D80510729/)

cc digantdesai freddan80 per zingo oscarandersson8218

[ghstack-poisoned]

Ninja91 committed:
Update on "Arm backend: Add 16A8W support and test for sigmoid operation"


Add 16A8W quantization support and test for the sigmoid operation in ExecutorTorch ARM backend.

This follows the pattern established for linear and mul operations, extending int16 support to sigmoid operations.

Changes:
- Add INT16 dtype validation support in op_sigmoid.py
- Add test_sigmoid_tensor_16a8w_tosa_INT test function
- Enable test_sigmoid.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80510729](https://our.internmc.facebook.com/intern/diff/D80510729/)

cc digantdesai freddan80 per zingo oscarandersson8218

[ghstack-poisoned]

Ninja91 committed:
Update on "Arm backend: Add 16A8W support and test for sigmoid operation"


Add 16A8W quantization support and test for the sigmoid operation in ExecutorTorch ARM backend.

This follows the pattern established for linear and mul operations, extending int16 support to sigmoid operations.

Changes:
- Add INT16 dtype validation support in op_sigmoid.py
- Add test_sigmoid_tensor_16a8w_tosa_INT test function
- Enable test_sigmoid.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80510729](https://our.internmc.facebook.com/intern/diff/D80510729/)

cc digantdesai freddan80 per zingo oscarandersson8218

[ghstack-poisoned]

Ninja91 committed:
Update on "Arm backend: Add 16A8W support and test for sigmoid operation"


Add 16A8W quantization support and test for the sigmoid operation in ExecutorTorch ARM backend.

This follows the pattern established for linear and mul operations, extending int16 support to sigmoid operations.

Changes:
- Add INT16 dtype validation support in op_sigmoid.py
- Add test_sigmoid_tensor_16a8w_tosa_INT test function
- Enable test_sigmoid.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80510729](https://our.internmc.facebook.com/intern/diff/D80510729/)

cc digantdesai freddan80 per zingo oscarandersson8218

[ghstack-poisoned]
